src/services/QueryExecutor.ts

```diff
@@ -32,29 +33,59 @@ export default class QueryExecutor {
    * Executes the queries found in the queries directory
    * and returns the result
    *
-   * @return {Promise<Object[]>}
+   * @param {IGraphqlVariables} [variables]
+   * @return {*}  {Promise<Object[]>}
    * @memberof QueryExecutor
    */
-  public async execQueries(): Promise<Object[]> {
-    await this.isLoadingSchema;
-    const graphqlPromises: Array<Promise<ExecutionResult>> = [];
-    for (const query of this.queries2Exec) {
-      this.logger.debug(`Executing ${query} query`);
-      const queries = (
-        await fs.readFile(`./queries/${this.name}/${query}`)
-      ).toString();
-      const queryPromise = graphql(this.loadedSchema, queries);
-      queryPromise
-        .then(() => {
-          this.logger.debug(`${query} executed successfully`);
-        })
-        .catch((e) => {
-          this.logger.error(`Failed executing ${query} query with`, e);
-        });
-      graphqlPromises.push(queryPromise);
-    }
-    const results = await Promise.all(graphqlPromises);
-    return results.map(result => result.data as Object);
+  public execQueries(variables?: IGraphqlVariables): Promise<Object[]> {
```

Comment:
The complexity here increases already quite much. Probably worth to split this method now or at least later on up in smaller private ones.